plugins/premium/powerpaste.md

```diff
@@ -95,48 +95,105 @@ tinymce.init({
 
 This option controls how content pasted from Microsoft Word is filtered.
 
-**Type:** `String`
+**Type:** `String` or `function`
 
 **Default value:** `prompt`
 
-The supported values are:
+The supported `string` values are:
 
-* `clean` - Preserve the structure of the content such as headings, tables and lists but remove inline styles and classes. This results in simple content that uses the site's CSS stylesheet while retaining the semantic structure from the original document.
+* `clean` - Preserve the structure of the content such as headings, tables, and lists but remove inline styles and classes. This results in simple content that uses the site's CSS stylesheet while retaining the semantic structure from the original document.
 * `merge` - Preserve the inline formatting and structure of the original document. Invalid and proprietary styles, tags and attributes are still removed ensuring that the HTML is valid while more closely matching the original document formatting.
-* `prompt` - Prompt the user to choose between the clean and merge options after attempting to paste Microsoft Word content.
+* `prompt` - Prompt the user to choose between the clean and merge options after attempting to paste HTML content.
+
+Alternatively, this option can take an asyncronous callback that returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that resolves to `clean` or `merge`. Allows the integrator to dynamically choose the paste mode on each paste event. Can be used to replicate the `prompt` dialog with custom, non-TinyMCE UI. Note: this requires [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) support.
```

Comment:
Tyler wanted this separate to the above list.

Reply:
Also, does this explanation make sense?

And: should I make the `Note...` an actual note, or is this fine?

Reply:
And and: two links to the MDN Promise docs, or just the first time?

Reply:
The last sentence honestly feels out of place... I'd remove it entirely. Additionally there's something that "smells" about the whole paragraph. I think it's that in the first sentence we repeat "that" and the second/third sentence starts with `Allows` and `Can` respectively which feels like there's content missing.

Will try to think of a suggestion for this later.

Reply:
I also noticed there's a typo in `asyncronous` here. Additionally, this is really not an async callback as that's something where by a callback function is passed that you need to call to continue (see https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Introducing#async_callbacks). As such, this is really just an "asynchronous function" instead.

Reply:
```suggestion
Alternatively, this option can take an asynchronous function that returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) which will resolve with `clean` or `merge`. This allows the paste mode to be dynamically chosen on each paste event. It can be used to replicate the `prompt` dialog with custom, non-TinyMCE UI for example.
```
Maybe?

Reply:
Included this suggestion in another commit